codegen/gateway.go

```diff
@@ -362,10 +398,29 @@ func NewEndpointSpec(
 		)
 	}
 
-	// TODO(sindelar): Use structures and validate middleware are defined in codebase.
-	middlewareStrings := make([]string, len(endpointConfigObj["middlewares"].([]interface{})))
+	// TODO(sindelar): Validate middleware are defined in codebase.
+	middlewares := make([]MiddlewareSpec, len(endpointConfigObj["middlewares"].([]interface{})))
 	for idx, middleware := range endpointConfigObj["middlewares"].([]interface{}) {
```

Comment:
there are several type assertions here, if anyone fails, the program panic. it is not ideal: this function is supposed to throw an error.

Use assertion like
`middlewares, ok := endpointConfigObj["middlewares"].([]interface{})` and throw specific error if `!ok`